File: problems/C++/0030_to_lower_case.cpp
// Problem Statement

// Implement function ToLowerCase() that has a string parameter str, and returns the same string in lowercase.

#include <bits/stdc++.h>

using namespace std;


class Solution {
public:
  string toLowerCase(string str) {

      string to_return;

      for(auto c:str){
          if(c>='A' && c<='Z')
              c='a'+c-'A';

          to_return.push_back(c);
      }

      return to_return;
  }
};


int main(){
  string str = "CODETOLOWER";
  Solution sol;
  cout<<sol.toLowerCase(str)<<endl;
  return 0;
}

File: Kotlin/problems/0030_to_lower_case.kt
// Problem Statement

// Implement function ToLowerCase() that has a string parameter str, and returns the same string in lowercase.
class Solution {
    fun toLowerCase(str: String): String {
        var result:String=""
        for(i in 0 until str.length){
            var letter = str[i]
            if(letter>='A' && letter<='Z'){
                letter = 'a'+(letter-'A')
            }
            result += letter
        }
        
        return result
    }
}

fun main(args: Array<String>) {
    val sol = Solution()
    val str = "LOVELY"
    println(sol.toLowerCase(str))
}